julia-mode.el

```diff
@@ -363,6 +365,7 @@ Based on `python-syntax-stringify'."
         (put-text-property string-start-pos (1+ string-start-pos)
                            'syntax-table (string-to-syntax "|"))))))
 
+(eval-when-compile (defvar julia-syntax-propertize-function))
 (unless (< emacs-major-version 24)
```

Comment:
There was a discussion about this on another pull request.  I think @yuyichao wants to branch to support older emacses

Reply:
@vspinu When setting the `byte-compile-error-on-warn`, will every compiler warning will fail the build?  I'm not sure how to silence the 

> `make-variable-buffer-local` not called at toplevel

warning from defining a minor mode in a condition.  Do you have any idea about that?

Reply:
Just wrap the whole `define-minor-mode` into `with-no-warning`. It will silence all warnings form that definition. No need for those empty LaTeX defvars either.

Reply:
> I think @yuyichao wants to branch to support older emacses

I just want to support the emacs that ship with centos 6. I have no idea how different different emacs versions are so it's up to you to decide what's the best way to support both. (With at least a best effort to support improvement that doesn't rely on new emacs features in both emacs version).

If you think branching with occassional backport is easier, feel free to create a new branch for that. I don't think you can create a PR for a new branch so just let me know and I'll pull the branch.